schemas/diagnostic_algorithms.rnc

```diff
@@ -100,6 +100,55 @@ internal_algorithm =
       element algorithm {
          attribute name { "Internal" },
          attribute material_phase_support { "multiple" }
+      }|
+      ## This diagnostic is derived from particle attributes, it is calculated
+      ## within the Particle_Diagnostics.F90 routine, and set via the ratio method.
+      element algorithm {
+         attribute name { "from_particles" },
+         attribute material_phase_support { "multiple" },
+	 (
+            ## Particle group used for calculation of diagnostic field.
+            element particle_group {
+	       attribute name { xsd:string },
+	       (
+	          ## Particle attribute used for calculation of diagnostic field.
+	          element particle_attribute {
+	             attribute name { xsd:string }
+	          }|
+	          ## Particle attribute_array used for calculation of diagnostic field.
+	          element particle_attribute_array {
+	             attribute name { xsd:string },
+		     ## Integer value corresponding to attribute position in array (starts from 1).
+		     element integer_val {
```

Comment:
Should be `integer_value` to be consistent with rest of the schema

Reply:
integer_value appears to only be used for cases with a set selection of options (e.g. when selecting dim). I've changed it to attribute_int instead for clarity that it's a user set parameter.

Reply:
Do you mean it didn't work when you tried (because diamond or spud treats `integer_value` specially?). Sorry to be pedantic on this but `attribute_int` isn't a good alternative. The main reason I objected to `integer_val` was that it abbreviates value - and we have tried very hard to avoid these in the schema. If `integer_value` doesn't work, maybe `attribute_index` ?

Also (final request I promise :crossed_fingers: ), I see there's tabs in various places you've edited the schema - could you please change these to spaces (use an editor that shows you tabs vs. spaces). Tabs only work if you use them consistently everywhere.